src/pip/_internal/commands/install.py

```diff
@@ -732,3 +737,25 @@ def create_os_error_message(error, show_traceback, using_user_site):
         parts.append(".\n")
 
     return "".join(parts).strip() + "\n"
+
+
+def prepare_sudo_warning():
+    # type: () -> None
+    """Output a warning for sudo users on Unix.
+
+    On Windows, users may run pip as Administrator without issues.
+    """
+    try:
+        if sys.platform == 'darwin' or sys.platform == 'linux':
+            warn_sudo = os.getuid() == 0
+        elif sys.platform == 'win32' or sys.platform == 'cygwin':
+            warn_sudo = False
+    except AttributeError:
+        warn_sudo = False
+
+    if warn_sudo and not running_under_virtualenv():
+        sudo_warning = ("Running pip as root is not recommended! "
```

Comment:
I wanna wordsmith on this, so bear with me. I don't think we need to tell users to use --user since pip does do that by default now (if site-packages is not writable). Instead, we should guide users toward using virtual environments.

```
Running pip as root is dangerous. It is possible to break packages installed for your operating system or a different package manager. You should use a virtual environment instead, and run pip as a non-root user.
```

Reply:
I also don't think this needs to be a separate variable. Please inline it with the call, and use the black code style here (we're slowly moving toward adopting that across the codebase).

```
logger.warning(
    "..."
    "..."
)
```

Reply:
Good point on suggesting virtual environments over --user. I've updated the warning.

My only nit is with the last phrase: `You should use a virtual environment instead, and run pip as a non-root user.` It's a bit vague with why and when to use virtual environments. Here's what I suggest:

`Consider using a virtual environment rather than trying to run as root. This may help resolve permission issues.`